moveit_plugins/moveit_ros_control_interface/src/controller_manager_plugin.cpp

```diff
@@ -370,35 +373,96 @@ class Ros2ControlManager : public moveit_controller_manager::MoveItControllerMan
     return c;
   }
 
+  static void simplifyControllerActivationDeactivation(std::vector<std::string>& activate_controllers,
+                                                       std::vector<std::string>& deactivate_controllers)
+  {
+    // Activation/deactivation is expected to be disjoint. For example, if controller B is a dependency of A (A chains
+    // to B) but controller B is also a dependency of C (B chains to B), then the switch from A->B to C->B would cause B
+    // to be in both the activation and deactivate list. This causes ROS2 control to through an error and reject the
+    // switch. The simplifyControllerActivationDeactivation function adds the logic needed to avoid this from happening.
```

Comment:
The rest of this file seems to put this in the doxygen for the function

Reply:
Added doxygen